apps/web/src/pages/api/onramp/provider-availability.ts

```diff
@@ -0,0 +1,21 @@
+import { ONRAMP_API_BASE_URL } from 'config/constants/endpoints'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const xForwardedFor = req.headers['x-forwarded-for']
+    const userIp = xForwardedFor ? (xForwardedFor as string).split(',')[0].trim() : null
+
+    const response = await fetch(`${ONRAMP_API_BASE_URL}/fetch-provider-availability-get?userIp=${userIp}`, {
```

Comment:
why are we adding this request proxy instead of doing the check on the service end?

Reply:
The service should be able to parse the request header, we don't need this api totally

Reply:
this has been resolved. added ip to the buy crypto api